packages/core/src/network.js

```diff
@@ -237,7 +270,10 @@ export class Network {
   // Called when a request has finished loading which triggers the this.onrequestfinished
   // callback. The request should have an associated response and be finished with any redirects.
   _handleLoadingFinished = async event => {
-    let request = this.#requests.get(event.requestId);
+    let { requestId } = event;
+    // wait for upto 2 seconds or check if response has been sent
+    await Promise.any([this.#requestsLifeCycleHandler.get(requestId).checkResponseSent, new Promise((resolve) => setTimeout(resolve, 2000))]);
```

Comment:
shouldnt _handleLoadingFailed wait for an event as well ?

Reply:
Yep, will add these in the next iteration.
Wanted to get a thumbs up on the implementation before proceeding further.